File: 0217-contains-duplicate/0217-contains-duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    const logger = new Set();
    for (let num of nums) {
        if (logger.has(num)) return true;
        logger.add(num);
    };
    return false;
};

File: 0217-contains-duplicate/0217-contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        if (nums.size() < 2) return false;

        set<int> logger;
        int i = 0;

        while (i < nums.size()) {
            if (logger.count(nums[i]) > 0) return true;
            logger.insert(nums[i]);
            i++;
        }

        return false;
    }
};